apps/organizations/models.py

```diff
@@ -0,0 +1,91 @@
+"""
+Declare and configure the models related to Organizations
+"""
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from cms.extensions import PageExtension
+from cms.extensions.extension_pool import extension_pool
+
+ORGANIZATIONS_PAGE_REVERSE_ID = 'organizations'
+
+
+class Organization(models.Model):
+    """
+    The Organization model is used to represent and record entities that manage courses.
+    It could be a university or a training company for example.
+
+    This model should be used to record structured data about the organization whereas the
+    associated page object is where we record the less structured information to display on the
+    page to present the organization.
+    """
+    name = models.CharField(_("name"), max_length=255)
+    code = models.CharField(
+        _("code"),
+        db_index=True,
+        max_length=100,
+        unique=True,
+        null=True,
+        blank=True,
+    )
+    logo = models.ImageField(
+        upload_to='organizations/logo/',
+        verbose_name=_("organization logo"),
+        help_text=_("Recommended size: 180x100"),
+        blank=True,
+    )
+
+    class Meta:
+        verbose_name = _('organization')
+
+    def __str__(self):
+        """Human representation of an organization"""
+        return '{model}: {name} ({code})'.format(
+            code=self.code,
+            name=self.name,
+            model=self._meta.verbose_name.title(),
+        )
+
+    # pylint: disable=arguments-differ
+    def save(self, *args, **kwargs):
+        """
+        Try to normalize the code a little for better uniqueness:
+            - all upper case,
+            - strip leading and trailing spaces,
+            - replace other spaces by a hyphen.
+        TODO: improve uniquess (find related Django blog post)
+        """
+        if self.code:
+            self.code = self.code.upper().strip().replace(' ', '-')
+        return super().save(*args, **kwargs)
+
+    def get_page(self):
+        """Return the draft CMS page"""
+        try:
+            return self.organization_pages.\
+                select_related('extended_object').\
+                get(extended_object__publisher_is_draft=True).\
+                extended_object
+        except OrganizationPage.DoesNotExist:
```

Comment:
Shouldn't we log something here?

Reply:
No it's not abnormal if there is no page. I use the try/except pattern because it is better to ask for forgiveness than permission.